tests/unit/mesh/test_exodus_writer.cpp

```diff
@@ -0,0 +1,252 @@
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+
+using namespace mfem;
+
+#ifdef MFEM_USE_NETCDF
+static void CompareMeshes(Mesh & mesh1, Mesh & mesh2)
+{
+   REQUIRE(mesh1.GetNE() == mesh2.GetNE());
+   REQUIRE(mesh1.GetNV() == mesh2.GetNV());
+   REQUIRE(mesh1.GetNBE() == mesh2.GetNBE());
+   REQUIRE(mesh1.GetNFaces() == mesh2.GetNFaces());
+
+   const FiniteElementSpace * fespace1 = mesh1.GetNodalFESpace();
+   const FiniteElementSpace * fespace2 = mesh2.GetNodalFESpace();
+
+   // Check elements.
+   Array<int> element_faces1, element_faces2;
+   Array<int> element_orient1, element_orient2;
+   Array<int> dofs1, dofs2;
+
+   for (int ielement = 0; ielement < mesh1.GetNE(); ielement++)
+   {
+      int attr1 = mesh1.GetAttribute(ielement);
+      int attr2 = mesh2.GetAttribute(ielement);
+
+      REQUIRE(attr1 == attr2);
+
+      Element::Type type1 = mesh1.GetElementType(ielement);
+      Element::Type type2 = mesh2.GetElementType(ielement);
+
+      REQUIRE(type1 == type2);
+
+      mesh1.GetElementFaces(ielement, element_faces1, element_orient1);
+      mesh2.GetElementFaces(ielement, element_faces2, element_orient2);
+
+      REQUIRE(element_faces1 == element_faces2);
+      REQUIRE(element_orient1 == element_orient2);
+
+      if (fespace1 && fespace2)
+      {
+         fespace1->GetElementDofs(ielement, dofs1);
+         fespace2->GetElementDofs(ielement, dofs2);
+      }
+      else
+      {
+         mesh1.GetElementVertices(ielement, dofs1);
+         mesh2.GetElementVertices(ielement, dofs2);
+      }
+
+      REQUIRE(dofs1 == dofs2);
+   }
+
+   // Check bdr elements.
+   for (int ibdr_element = 0; ibdr_element < mesh1.GetNBE(); ibdr_element++)
+   {
+      int attr1 = mesh1.GetBdrAttribute(ibdr_element);
+      int attr2 = mesh2.GetBdrAttribute(ibdr_element);
+
+      REQUIRE(attr1 == attr2);
+
+      Element::Type type1 = mesh1.GetBdrElementType(ibdr_element);
+      Element::Type type2 = mesh2.GetBdrElementType(ibdr_element);
+
+      REQUIRE(type1 == type2);
+
+      int face_index1 = mesh1.GetBdrElementFaceIndex(ibdr_element);
+      int face_index2 = mesh2.GetBdrElementFaceIndex(ibdr_element);
+
+      REQUIRE(face_index1 == face_index2);
+   }
+
+   // Check face vertices.
+   Array<int> face_vertices1, face_vertices2;
+   for (int iface_index = 0; iface_index < mesh1.GetNFaces(); iface_index++)
+   {
+      mesh1.GetFaceVertices(iface_index, face_vertices1);
+      mesh2.GetFaceVertices(iface_index, face_vertices2);
+
+      REQUIRE(face_vertices1 == face_vertices2);
+   }
+}
+#endif
+
+TEST_CASE("ExodusII Write Hex8", "[Mesh]")
+{
+#ifdef MFEM_USE_NETCDF
+   // Load Exodus II mesh from file.
+   std::string fpath_original = "data/simple-cube-hex8.e";
+   Mesh original_mesh = Mesh::LoadFromFile(fpath_original, 0, 0, true);
+
+   // Write generated Exodus II mesh to file.
+   std::string fpath_generated = "data/simple-cube-hex8-out.e";
```

Comment:
Two small points.

First, we don't typically write in the `data` directory. This is a minor point but it may be better to place these temporary output files in the current directory.

Second, the test should remove its temporary files to avoid clutter and false positives in future tests. For example, this is done in a similar unit test using `remove` as seen here: https://github.com/mfem/mfem/blob/41a40ebf57e982c54eaaaab514b487c4182ae195/tests/unit/fem/test_datacollection.cpp#L137